### DISCUSSION TITLE
steam: --private= home dotfiles pollution

### DISCUSSION THREAD

**Discussion by kuesji** [netblue30/firejail#5048] [Q&A] [answered]
i am using `private` to specify a directory for home in profile file like `private /data/jails/kuesji/steam` and all steam related files goes to this directory but whitelisted directories in steam profile ends up my home directory too.

how i can stop this? i don't wanna see any steam ( or any sandboxed program ) files/directories ( even whitelisted ones ) in my home directory or anywhere outiside of it's specified home directory.

also:
- directories created by whitelist rules in my home is all empty.
- this directories recreated at start of sandbox if i delete them by hand
  
i thought it's a bug in blacklist/whitelist mechanism but i'm not sure. this is why i asked in there.

demonstration picture at https://imgur.com/a/rWWviMC ( left is sandbox, right is my home )

**Comment by rusty-snake** [comment, marked as answer]
It's a bug <sub>(with a number <1000 :speak_no_evil: )</sub>: #903

To workaround, add `ignore mkdir`.

**Comment by kuesji** [comment]
thanks, it's worked. looks like issue is old but i didn't notice before because i started using steam recently. 


also i saw `... to /etc/firejail/globals.local and create files manually, or run applications once outside Firejail, which is usually sufficient to get all required directories and files.` sentence at linked issue and i'm curious. 
is `ignore mkfile` and `ignore mkdir` have any side effect which will create more issues in future?

**Reply by rusty-snake** [reply]
As long as you keep using `private`, no.
